### PR TITLE
Creates the list group component and bumps springer lists components

### PR DESCRIPTION
**Pull request by benjclark**
I have added this list-group component in to the springer-lists package as it seemed as though it was named with the intention of being a package for all springer lists? Which does seem like a decent idea

### CODE REVIEW

**Comment by benjclark** [comment]
Do you think it's acceptable to create a dependency on `@mixin interface-link` from the springer-context package in this component?

File: packages/springer-lists/scss/50-components/list-group.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	line-height: $list-group-line-height;
+}
+
+.c-list-group__item {
+	> a {
+		@include interface-link;
+	}
+
+	.c-list-group--sm & {
+		padding: $list-group-padding-sm;
```
Comment (davidpauljunior) [inline packages/springer-lists/scss/50-components/list-group.scss]: These should be double `--` after the component name.
`$list-group--padding-sm`

File: packages/springer-lists/scss/50-components/list-group.scss [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	@include u-list-reset;
+	line-height: $list-group-line-height;
+}
+
+.c-list-group__item {
+	> a {
+		@include interface-link;
+	}
+
+	.c-list-group--sm & {
```
Comment (davidpauljunior) [inline packages/springer-lists/scss/50-components/list-group.scss]: I think it was agreed in previous open-space that it's preferable to not use this sort of nesting.
So `.c-list-group--sm .c-list-group__item`.

File: packages/springer-lists/package.json [hunk 3]
```diff
@@ -1,12 +1,12 @@
 {
     "name": "@springernature/springer-lists",
-    "version": "0.0.0",
+    "version": "0.0.2",
```
Comment (davidpauljunior) [inline packages/springer-lists/package.json]: I guess this should be `0.0.1`, even though that's arguably what the previous version should have been.

**Comment by davidpauljunior** [comment]
Yeah, it's not a "dependency" on `@mixin interface-link`, just using it and it's defined in the context so it's a mandatory part of any springer app.

File: packages/springer-lists/package.json (hunk 3 above)
Comment (davidpauljunior) [reply packages/springer-lists/package.json]: Oh I just realised what happened here.  Previously the entry was added to the history but not to the package

**Pull request merged**